src/main/java/de/cyface/collector/verticle/CollectorApiVerticle.java

```diff
@@ -208,8 +208,8 @@ private void setupRoutes(final String publicKey, final String privateKey, final
                 .failureHandler(new AuthenticationFailureHandler());
         // Set up data collector route
         v2ApiRouter.post("/measurements").handler(BodyHandler.create().setDeleteUploadedFilesOnEnd(false))
-                .handler(JWTAuthHandler.create(jwtAuthProvider)).handler(new MeasurementHandler())
-                .failureHandler(new AuthenticationFailureHandler());
```

Comment:
Was this what you were talking about? That the auth failure handler was not really called so you commented it out?

Reply:
This causes the system to always return status 401 even for internal server errors or cases where 422 is appropriate. The AuthenticationFailureHandler should only be used for the initial auth call. On other calls the JWTAuthHandler from vertx provides the correct status codes on failure.